Source/Diagnostics/FlushFormats/FlushFormatOpenPMD.H

```diff
@@ -0,0 +1,40 @@
+#ifndef WARPX_FLUSHFORMATOPENPMD_H_
+#define WARPX_FLUSHFORMATOPENPMD_H_
+
+#include "FlushFormat.H"
+#include "Diagnostics/WarpXOpenPMD.H"
+
+/**
+ * \brief This class aims at dumping diags data to disk using the OpenPMD standard.
+ * In particular, function WriteToFile takes fields and particles as input arguments,
+ * and writes data to file.
+ */
+class FlushFormatOpenPMD : public FlushFormat
+{
+public:
+
+    /** Constructor takes name of diagnostics to set the output directory */
+    FlushFormatOpenPMD (const std::string& diag_name);
+
+    /** Flush fields and particles to plotfile */
+    virtual void WriteToFile (
+        const amrex::Vector<std::string> varnames,
+        const amrex::Vector<const amrex::MultiFab*> mf,
+        amrex::Vector<amrex::Geometry>& geom,
+        const amrex::Vector<int> iteration, const double time,
+        MultiParticleContainer& mpc, int nlev, const std::string prefix,
+        bool plot_raw_fields,
+        bool plot_raw_fields_guards,
+        bool plot_raw_rho, bool plot_raw_F) const override final;
+
+    ~FlushFormatOpenPMD ();
+
+private:
+    /** Which backend to use (ADIOS, ADIOS2 or HDF5). Default depends on what is available */
+    std::string openpmd_backend {"default"};
+    bool openpmd_tspf = true; //!< one file per timestep (or one file for all steps)
```

Comment:
Should this be prefixed with `m_`?
```suggestion
    std::string m_openpmd_backend {"default"};
    bool m_openpmd_tspf = true; //!< one file per timestep (or one file for all steps)
```

Reply:
Agreed.